rust/src/lib.rs

```diff
@@ -1,3 +1,11 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(dead_code)]
```

Comment:
funny smell here.
I guess some code is only exposed via std or no-std?
We could also just conditionally expose it.

Reply:
Yes, compile by using `#![cfg_attr(not(feature = "std"), no_std)]` to conditionally select between `std` and `no_std`.

Reply:
I was commenting on `#![allow(dead_code)]`, which I removed in another PR